diff --git a/LAVA-M/base64/coreutils-8.24-lava-safe/src/base64.c b/LAVA-M/base64/coreutils-8.24-lava-safe/src/base64.c
index 2ff2956..f173595 100644
--- a/LAVA-M/base64/coreutils-8.24-lava-safe/src/base64.c
+++ b/LAVA-M/base64/coreutils-8.24-lava-safe/src/base64.c
@@ -12,6 +12,7 @@ unsigned int lava_get(unsigned int bug_num) {
     if (0x6c617661 - bug_num == lava_val[bug_num] ||
         SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
         dprintf(1, "Successfully triggered bug %d, crashing now!\n", bug_num);
+        fprintf(stderr, "\nLAVALOG: %d, %s: %d\n", bug_num, __FILE__, __LINE__);
     }
     else {
         //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);                                                                                                                                                                         
diff --git a/LAVA-M/base64/validate.sh b/LAVA-M/base64/validate.sh
index 29c8f0a..d94fae8 100755
--- a/LAVA-M/base64/validate.sh
+++ b/LAVA-M/base64/validate.sh
@@ -9,13 +9,13 @@ echo "Building buggy ${PROG}..."
 
 cd coreutils-8.24-lava-safe
 make clean &> /dev/null
-./configure --prefix=`pwd`/lava-install LIBS="-lacl" &> /dev/null
+./configure --prefix=`pwd`/lava-install &> /dev/null
 make -j $(nproc) &> /dev/null
 make install &> /dev/null
 cd ..
 
 echo "Checking if buggy ${PROG} succeeds on non-trigger input..."
-./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null 
+./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null
 rv=$?
 if [ $rv -lt 128 ]; then
     echo "Success: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
@@ -23,13 +23,3 @@ else
     echo "ERROR: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
 fi
 
-echo "Validating bugs..."
-cat validated_bugs | while read line ; do
-    INPUT_FUZZ=$(printf "$INPUT_PATTERN" $line)
-    { ./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_FUZZ} ; } &> /dev/null
-    echo $line $?
-done > validated.txt
-
-awk 'BEGIN {valid = 0} $2 > 128 { valid += 1 } END { print "Validated", valid, "/", NR, "bugs" }' validated.txt
-
-echo "You can see validated.txt for the exit code of each buggy version."
diff --git a/LAVA-M/md5sum/coreutils-8.24-lava-safe/src/md5sum.c b/LAVA-M/md5sum/coreutils-8.24-lava-safe/src/md5sum.c
index f220f24..072faeb 100644
--- a/LAVA-M/md5sum/coreutils-8.24-lava-safe/src/md5sum.c
+++ b/LAVA-M/md5sum/coreutils-8.24-lava-safe/src/md5sum.c
@@ -10,6 +10,7 @@ unsigned int lava_get(unsigned int bug_num) {
     if (0x6c617661 - bug_num == lava_val[bug_num] ||
         SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
         dprintf(1,"Successfully triggered bug %d, crashing now!\n", bug_num);
+        fprintf(stderr, "\nLAVALOG: %d, %s: %d\n", bug_num, __FILE__, __LINE__);
     }
     else {
         //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);
diff --git a/LAVA-M/md5sum/validate.sh b/LAVA-M/md5sum/validate.sh
index cbb0223..7c21faa 100755
--- a/LAVA-M/md5sum/validate.sh
+++ b/LAVA-M/md5sum/validate.sh
@@ -9,13 +9,13 @@ echo "Building buggy ${PROG}..."
 
 cd coreutils-8.24-lava-safe
 make clean &> /dev/null
-./configure --prefix=`pwd`/lava-install LIBS="-lacl" &> /dev/null
+./configure --prefix=`pwd`/lava-install &> /dev/null
 make -j $(nproc) &> /dev/null
 make install &> /dev/null
 cd ..
 
 echo "Checking if buggy ${PROG} succeeds on non-trigger input..."
-./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null 
+./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null
 rv=$?
 if [ $rv -lt 128 ]; then
     echo "Success: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
@@ -23,13 +23,3 @@ else
     echo "ERROR: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
 fi
 
-echo "Validating bugs..."
-cat validated_bugs | while read line ; do
-    INPUT_FUZZ=$(printf "$INPUT_PATTERN" $line)
-    { ./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_FUZZ} ; } &> /dev/null
-    echo $line $?
-done > validated.txt
-
-awk 'BEGIN {valid = 0} $2 > 128 { valid += 1 } END { print "Validated", valid, "/", NR, "bugs" }' validated.txt
-
-echo "You can see validated.txt for the exit code of each buggy version."
diff --git a/LAVA-M/uniq/coreutils-8.24-lava-safe/src/uniq.c b/LAVA-M/uniq/coreutils-8.24-lava-safe/src/uniq.c
index 8edb5a5..ab68251 100644
--- a/LAVA-M/uniq/coreutils-8.24-lava-safe/src/uniq.c
+++ b/LAVA-M/uniq/coreutils-8.24-lava-safe/src/uniq.c
@@ -10,6 +10,7 @@ unsigned int lava_get(unsigned int bug_num) {
     if (0x6c617661 - bug_num == lava_val[bug_num] ||
         SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
         dprintf(1, "Successfully triggered bug %d, crashing now!\n", bug_num);
+        fprintf(stderr, "\nLAVALOG: %d, %s: %d\n", bug_num, __FILE__, __LINE__);
     }
     else {
         //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);
diff --git a/LAVA-M/uniq/validate.sh b/LAVA-M/uniq/validate.sh
index 2ba37c8..f8577fb 100755
--- a/LAVA-M/uniq/validate.sh
+++ b/LAVA-M/uniq/validate.sh
@@ -9,13 +9,13 @@ echo "Building buggy ${PROG}..."
 
 cd coreutils-8.24-lava-safe
 make clean &> /dev/null
-./configure --prefix=`pwd`/lava-install LIBS="-lacl" &> /dev/null
+./configure --prefix=`pwd`/lava-install &> /dev/null
 make -j $(nproc) &> /dev/null
 make install &> /dev/null
 cd ..
 
 echo "Checking if buggy ${PROG} succeeds on non-trigger input..."
-./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null 
+./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null
 rv=$?
 if [ $rv -lt 128 ]; then
     echo "Success: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
@@ -23,13 +23,3 @@ else
     echo "ERROR: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
 fi
 
-echo "Validating bugs..."
-cat validated_bugs | while read line ; do
-    INPUT_FUZZ=$(printf "$INPUT_PATTERN" $line)
-    { ./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_FUZZ} ; } &> /dev/null
-    echo $line $?
-done > validated.txt
-
-awk 'BEGIN {valid = 0} $2 > 128 { valid += 1 } END { print "Validated", valid, "/", NR, "bugs" }' validated.txt
-
-echo "You can see validated.txt for the exit code of each buggy version."
diff --git a/LAVA-M/who/coreutils-8.24-lava-safe/src/who.c b/LAVA-M/who/coreutils-8.24-lava-safe/src/who.c
index b5d6fd3..2d20d8f 100644
--- a/LAVA-M/who/coreutils-8.24-lava-safe/src/who.c
+++ b/LAVA-M/who/coreutils-8.24-lava-safe/src/who.c
@@ -4,20 +4,7 @@ static unsigned int lava_val[1000000];
 void lava_set(unsigned int bug_num, unsigned int val);
 void lava_set(unsigned int bug_num, unsigned int val) { lava_val[bug_num] = val; }
 unsigned int lava_get(unsigned int bug_num);
-unsigned int lava_get(unsigned int bug_num) {
-#if 0
 #define SWAP_UINT32(x) (((x) >> 24) | (((x) & 0x00FF0000) >> 8) | (((x) & 0x0000FF00) << 8) | ((x) << 24))
-    if (0x6c617661 - bug_num == lava_val[bug_num] ||
-        SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
-        printf("Successfully triggered bug %d, crashing now!\n", bug_num);
-        exit(0);
-    }
-    else {
-        //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);
-    }
-#endif
-    return lava_val[bug_num];
-}
 /* GNU's who.
    Copyright (C) 1992-2015 Free Software Foundation, Inc.
 
@@ -61,6 +48,18 @@ unsigned int lava_get(unsigned int bug_num) {
 # include <grp.h>
 #endif
 
+unsigned int lava_get(unsigned int bug_num) {
+    if (0x6c617661 - bug_num == lava_val[bug_num] ||
+        SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
+        printf("Successfully triggered bug %d, crashing now!\n", bug_num);
+        fprintf(stderr, "\nLAVALOG: %d, %s: %d\n", bug_num, __FILE__, __LINE__);
+    }
+    else {
+        //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);
+    }
+    return lava_val[bug_num];
+}
+
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "who"
 
diff --git a/LAVA-M/who/validate.sh b/LAVA-M/who/validate.sh
index e2ae83e..8f95d66 100755
--- a/LAVA-M/who/validate.sh
+++ b/LAVA-M/who/validate.sh
@@ -9,13 +9,13 @@ echo "Building buggy ${PROG}..."
 
 cd coreutils-8.24-lava-safe
 make clean &> /dev/null
-./configure --prefix=`pwd`/lava-install LIBS="-lacl" &> /dev/null
+./configure --prefix=`pwd`/lava-install &> /dev/null
 make -j $(nproc) &> /dev/null
 make install &> /dev/null
 cd ..
 
 echo "Checking if buggy ${PROG} succeeds on non-trigger input..."
-./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null 
+./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_CLEAN} &> /dev/null
 rv=$?
 if [ $rv -lt 128 ]; then
     echo "Success: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
@@ -23,13 +23,3 @@ else
     echo "ERROR: ${PROG} ${PROGOPT} ${INPUT_CLEAN} returned $rv"
 fi
 
-echo "Validating bugs..."
-cat validated_bugs | while read line ; do
-    INPUT_FUZZ=$(printf "$INPUT_PATTERN" $line)
-    { ./coreutils-8.24-lava-safe/lava-install/bin/${PROG} ${PROGOPT} ${INPUT_FUZZ} ; } &> /dev/null
-    echo $line $?
-done > validated.txt
-
-awk 'BEGIN {valid = 0} $2 > 128 { valid += 1 } END { print "Validated", valid, "/", NR, "bugs" }' validated.txt
-
-echo "You can see validated.txt for the exit code of each buggy version."
